templates/airflow-ec2.j2

```diff
@@ -0,0 +1,338 @@
+                priority=100
+                autostart=true
+                autorestart=true
+                user=root
+
+
+          commands:
+            01_restart_supervisor:
+              command: !Join
+                - ''
+                - - 'systemctl restart supervisor;'
+
+    Properties:
+      ImageId: !Ref AMIId
+      InstanceType: !Ref InstanceType
+      # KeyName: !If [HasKeyName, !Ref KeyName, !Ref 'AWS::NoValue']
+      PropagateTagsToVolumeOnCreation: true
+      BlockDeviceMappings:
+        - DeviceName: '/dev/sda1'
+          Ebs:
+            DeleteOnTermination: true
+            VolumeSize: !Ref VolumeSize
+            Encrypted: !Ref EncryptVolume
+      SubnetId: !Ref VpcSubnet
+      SecurityGroupIds:
+        - !GetAtt VpnSecurityGroup.GroupId
+{% if sceptre_user_data.OpenPorts is defined %}
+        - !GetAtt InstanceSecurityGroup.GroupId
+{% endif %}
+      IamInstanceProfile: !Ref InstanceProfile
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash -xe
+
+          # Add the official Docker (apt-get) repository
+          mkdir -p /etc/apt/keyrings
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg \
+            | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+          echo \
+            "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+            $(lsb_release -cs) stable" \
+            | tee /etc/apt/sources.list.d/docker.list > /dev/null
+
+          # Install some tools using apt-get
+          apt-get update -y
+          apt-get install -y git
+          apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
+
+          # install supervisor
+          # - note using the distribution version because it is acceptably recent
+          #   and provides the integration into systemd
+          # - service config is placed at /usr/lib/systemd/system/supervisor.service
+          # - use option here to avoid overwriting the conf file
+          #   that is already in place at /etc/supervisor/supervisord.conf
+          apt-get install -o Dpkg::Options::="--force-confold" supervisor
+
+          # Clone Airflow DAGs repository
+          git clone https://github.com/Sage-Bionetworks-Workflows/orca-recipes-airflow.git /root/orca-recipes-airflow
+
+          # Install and run cfn-init
+          mkdir -p /opt/aws/bin
+          wget https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz
+          python3 -m easy_install --script-dir /opt/aws/bin aws-cfn-bootstrap-py3-latest.tar.gz
+          /opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource Ec2Instance --configsets SetupCfn,SetupTools,SetupAirflow --region ${AWS::Region}
+          /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource Ec2Instance --region ${AWS::Region}
+
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT15M
```

Comment:
Is the 15 minute timeout because it takes awhile to spin this up?

Reply:
Yes, the timeout was 5 minutes initially, but it was not making it in time.